cli/src/description_util.rs

```diff
@@ -42,23 +42,32 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
 
 pub fn combine_messages(
     repo: &ReadonlyRepo,
-    source: &Commit,
-    destination: &Commit,
+    commits: &[&Commit],
     settings: &UserSettings,
 ) -> Result<String, CommandError> {
-    let description = if source.description().is_empty() {
-        destination.description().to_string()
-    } else if destination.description().is_empty() {
-        source.description().to_string()
-    } else {
-        let combined = "JJ: Enter a description for the combined commit.\n".to_string()
-            + "JJ: Description from the destination commit:\n"
-            + destination.description()
-            + "\nJJ: Description from the source commit:\n"
-            + source.description();
-        edit_description(repo, &combined, settings)?
-    };
-    Ok(description)
+    let non_empty = commits
+        .iter()
+        .filter(|c| !c.description().is_empty())
+        .take(2)
+        .collect_vec();
+    match non_empty.as_slice() {
+        &[] => {
+            return Ok(String::new());
+        }
+        &[commit] => {
+            return Ok(commit.description().to_owned());
+        }
+        _ => {}
+    }
+    // Produce a combined description with instructions for the user to edit.
+    // Include empty descriptins too, so the user doesn't have to wonder why they
+    // only see 2 descriptions when they combined 3 commits.
+    let mut combined = "JJ: Enter a description for the combined commit.".to_string();
+    for commit in commits {
+        combined.push_str("\nJJ: Description from one commit:\n");
```

Comment:
Since descriptions aren't chronologically sorted but `[destination, source..]`, it might be better to label them as "the destination", "source", ...